lib/Sema/TypeCheckCaptures.cpp

```diff
@@ -240,6 +250,103 @@ class FindCapturedVars : public ASTWalker {
       checkType(VD->getInterfaceType(), VD->getLoc());
   }
 
+  /// Emit warnings for capture list entries that:
+  ///   1.  Have weak/unowned reference ownership, but refer to a Decl that is
+  ///       _not_ weak/unowned.
+  ///   2.  Provide the only reference to such Decls withing the current scope
+  void
+  diagnoseImplicitWeakToStrongCaptureListBindings() { // TODO: where should a
+                                                      // method like this live?
```

Comment:
to clarify, by 'all this logic' to what are you referring exactly? i haven't entirely thought through the implications of moving different bits of this around, but i would like to avoid having to re-implement any of the closure capture compuatation algorithm if possible.